syntax = 'proto2';

message H2ModPacket {

	enum Type {
		set_player_team = 2;
		set_unit_grenades = 3;
		map_info_request = 4;
		set_lobby_settings = 5;
	}

	required H2ModPacket.Type type = 1;
	optional h2mod_set_team h2_set_player_team = 2;
	optional h2mod_set_grenade set_grenade = 3;
	optional h2mod_map_info map_info = 4;
	optional h2mod_lobby_settings lobby_settings = 5;
}

message h2mod_set_grenade {
	required uint32 type = 1;
	required uint32 count = 2;
	required uint32 pIndex = 3;
}

message h2mod_set_team {
	required uint32 peerIndex = 1;	
	required uint32 team = 2;
}

message h2mod_map_info {
	required string mapFileName = 1;
	required uint32 mapSize = 2;
}

message h2mod_lobby_settings {
	required uint32 mp_explosion_physics = 1;
	required uint32 mp_sputnik = 2;
	required uint32 mp_grunt_bday_party = 3;
	required uint32 grenade_chain_react = 4;
	required uint32 banshee_bomb = 5;
	required uint32 mp_blind = 6;
	required uint32 flashlight = 7;
}

message StringMap_FieldEntry {
  optional string key = 1;
  optional string value = 2;
}

message h2mod_map_checksum_state {
	required bool is_offical = 1;
	// switch to this if we update to version 3
	//map<string, string> map_checksum_list = 2;
	repeated StringMap_FieldEntry map_checksum_list = 2;
}
